roles/xbackbone/tasks/main.yml

```diff
@@ -0,0 +1,64 @@
+#########################################################################
+# Title:            Community: XBackBone                                #
+# Author(s):        Quick                                               #
+# URL:              https://github.com/Cloudbox/Community               #
+# Docker Image(s):  pe46dro/xbackbone-docker                            #
+# --                                                                    #
+#         Part of the Cloudbox project: https://cloudbox.works          #
+#########################################################################
+#                   GNU General Public License v3.0                     #
+#########################################################################
+---
+- name: "Set DNS Record on CloudFlare"
+  include_role:
+    name: cloudflare-dns
+  vars:
+    record: "{{ xbackbone.subdomain|default('xbackbone',true) }}"
+  when: cloudflare_enabled
+
+- name: Stop and remove any existing container
+  docker_container:
+    name: xbackbone
+    state: absent
+
+- name: Create xbackbone directories
+  file: "path={{ item }} state=directory mode=0777 owner=1000 group=1000 recurse=yes"
+  with_items:
+    - /opt/xbackbone/
+    - "{{ xbackbone.storage }}"
+    
+- name: Set '/opt/xbackbone' permissions recursively
+  shell: "sudo chown -R 1000:1000 /opt/xbackbone"
+  args:
+    warn: no
+
+- name: Create and start container
+  docker_container:
+    name: xbackbone
+    image: "pe46dro/xbackbone-docker"
+    pull: yes
+    published_ports:
+      - "127.0.0.1:90:80"
+    env:
+      TZ: "{{ tz }}"
+      DOMAIN: "{{ xbackbone.subdomain|default('xbackbone',true) }}.{{ xbackbone.domain | default(user.domain,true) }}"
+      URL: 'https:\/\/{{ xbackbone.subdomain|default("xbackbone",true) }}.{{ xbackbone.domain | default(user.domain,true) }}'
```

Comment:
This `https:\/\/` can be changed to just `https://` you don't seem to need to escape the \ in ansible as you do when running it manually like the xbackbone docker page suggested.

Reply:
It's not an issue with docker compose or ansible.    Xbackbone requires it to be that way. 

You can test it yourself, but it doesn't work without it when I tried.

Reply:
They're most likely for escaping slashes in shells.

Running the xbackbone command from their docker page

> docker run -p 80:80 \
    -e APP_NAME=XBackBone \
    -e URL=http:\/\/127.0.0.1 \
    --name xbb \
    pe46dro/xbackbone-docker

And then inspecting the created container you can see
`"URL=http://127.0.0.1",`

Running your role, and then inspecting the created container you can see
`"URL=https:\\/\\/xbackbone.domain.tld",`
Because yours is being passed through literally like that. Manually editing your role so it uses https:// works fine. Both ways work fine but no need to escape the slashes if you don't need to.